src/mappings/referrer.ts

```diff
@@ -17,54 +19,69 @@ export function handleReferralCodeCreated(event: OnReferralCodeCreated): void {
 }
 
 export function handleReferralCodeUpserted(event: OnReferralCodeUpserted): void {
-    let existingReferralCode = getReferralCode(event.params.oldReferralCode)
-    let trader = getOrCreateTrader(event.params.addr)
-    // Only proceed any action, if the referee has a code attached
-    if (existingReferralCode) {
-        // Referee removes code
-        if (event.params.action == 1) {
-            // trader no longer has a referee code
-            trader.refereeCode = null
-            trader.save()
-
-            // update referee count for code
-            existingReferralCode.numReferees = existingReferralCode.numReferees.minus(BI_ONE)
-            existingReferralCode.save()
-            return
-        }
+    if (!event.params.oldReferralCode) {
+        return
+    }
 
-        // add and update should only be possible if newReferralCode contains a value
-        if (event.params.newReferralCode != "") {
-            let newReferralCode = getReferralCode(event.params.newReferralCode)
-            let newReferralCodeDayData = getReferralCodeDayData(event, newReferralCode.id)
-            if (event.params.action == 2) {
-                // nothing changes here
-                if (event.params.oldReferralCode == event.params.newReferralCode) return
-            }
+    const oldRefCode = getReferralCode(event.params.oldReferralCode)
+    if (!oldRefCode) {
+        return
```

Comment:
I think we missed the case where a user upsert his ref code for the first time (case `ADD` in the source code). Case `ADD` is returned here since `oldReferralCode` is not found in subgraph.

I the original code, since `existingReferralCode` is always not null, case `ADD` is handled starting from line 38.